smart-contracts-tutorial/guess37-multiple-participants.md

```diff
@@ -0,0 +1,114 @@
+
+How does the game player know they are playing the `guess37` game with the actual contract source code defined in the [previous step](./first-contract)?
+What if the contract installer makes them play a different game?
+Let's see what guarantees the game player has even if they're suspicious of the contract installer
+
+The first and only assumption we need is that the game player trust the blockchain and the contract host and handoff service running on top of it.
+If the game player does not trust the contract host and blockchain it's running on, no security property can be guaranteed
+
+If they trust the contract host and blockchain it's running on, then they trust that `handoff` contains the actual handoff service
+If they trust it, they trust that `board` contains the actual board that the contract installer set up for them. The reason they can trust it is that, [by design, only the first person grabbing the board can have it](https://github.com/Agoric/ERTP/blob/91aab6abe0f0d9db61f0ac3b858ba6c310410aa5/more/handoff/handoff.js#L16-L27). So, if the game player retrieved the board, they know they have the right one and they are the only person with access to this specific board
+
+When `playerInvite = board~.lookup("invite")` happens, the game player knows that they retrieve what the contract installer shared with them.
+
+But how does the game player know the contract installer didn't just share with them some random object that acts like a contract seat but isn't really one? This is where `host~.redeem(playerInvite)` comes to play
+
+The actual seat is not handed directly by the contract installer. Instead, the game player redeems it from the trusted contract host running on the blockchain. This way, the game player know that what they have after `redeem` is an actual seat to a smart contact. A successful call to `host~.redeem` offers the following guarantees:
+- what the game player receives is an actual seat to a contract hosted on this contract host
+- the game player is the only person with access to this seat
+    - Anyone trying `host~.redeem(playerInvite)` a second time will receive an error. Even the contract installer! (notice the contract installer did not redeem the invite in this section, they only shared it through the handoff service). No matter how many people saw or shared the invite, if you redeem the seat, ERTP guarantees that you're the only one holding it
+
+So, the game player has a seat to an actual contract. Cool! But how can they know it's to the correct contract?
+They can use the `playerInvite` object for this purpose.
+
+`playerInvite` is a [payment object](https://github.com/Agoric/ERTP/blob/91aab6abe0f0d9db61f0ac3b858ba6c310410aa5/core/mint.chainmail#L136-L154)
+
+
+`playerInvite~.getBalance()` provides informations about the invite 
+
+```
+TODO
+Figure out how to:
+- guarantee this is a genuine invite produced by the contract host (via comparison with the invite assay?)
+- retrieve the source code and read it
```

Comment:
Not sure what you had in mind with this comment, but I wondered whether it would be useful to have the reader print out the contract text, either from host~.getInstallationSourceCode(installation), or from home.guess37ContractSource.

Reply:
Yes exactly, i was thinking `host~.getInstallationSourceCode(installation)`

`home.guess37ContractSource` is not part of ERTP and is a temporary hack for code upload, so i'm leaving it out of the explanation
Also, for the purpose of studying the invite and the underlying contract, i think it's more reliable to retrieve the source code from the (trusted) contract host since even the contract installer shouldn't be able to corrupt it